04-single-hop-6lowpan-icmp/test_spec04.py

```diff
@@ -12,13 +12,33 @@
 
 APP = 'examples/gnrc_networking'
 TASK10_APP = 'tests/gnrc_udp'
+PKTBUF_PROBE_DELAY = 5
+PKTBUF_PROBE_ATTEMPTS = 12
 pytestmark = pytest.mark.rc_only()
 
 
 class Shell(Ifconfig, GNRCICMPv6Echo, GNRCPktbufStats):
     pass
 
 
+def _probe_pktbuf(node):
+    print("check pktbuf on", node.riotctrl.env.get("PORT"))
+    return pktbuf(node).is_empty()
+
+
+def check_pktbuf(nodes):
+    """ Probe every PKTBUF_PROBE_DELAY seconds if the pktbuf of all nodes
+        is empty. If so, this function returns True.
+        After PKTBUF_PROBE_ATTEMPTS failed attempts, this function
+        returns False.
+    """
+    for _ in range(PKTBUF_PROBE_ATTEMPTS):
+        if not False in [_probe_pktbuf(n) for n in nodes]:
```

Comment:
Why the indirection to `_probe_pktbuf`? Just because of the `print`? Ideally this is not needed.

```suggestion
        if not False in [pktbuf(n).is_empty() for n in nodes]:
```